### ISSUE TITLE
Issues with Vulkan-Samples setup for Android development (vol 2)

### ISSUE THREAD

**Issue by iexavl** [KhronosGroup/Vulkan-Samples#855]
I am opening this issue because I am experiencing the exact same problem as the issue described in this thread:
https://github.com/KhronosGroup/Vulkan-Samples/issues/744
Unfortunately the issue ended up not being resolved. Here is the error:
vulkan_samples.app.main | debug | arm64-v8a Compiler exited with error code 1: C:\Users\voracity\AppData\Local\Android\Sdk\ndk\25.2.9519653\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe -xc++ --target=aarch64-none-linux-android30 --sysroot=C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot -DENABLE_HLSL -DENABLE_OPT=0 -DGLSLANG_OSINCLUDE_UNIX -DROOT_PATH_SIZE=37 -DVKB_DEBUG -ID:/gitroot/vulkanjava/Vulkan-Samples/build/android_gradle/app/build-native/Debug/6s723z10/arm64-v8a/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -DDEBUG=0 -fno-limit-debug-info -fPIC -Wall -Wuninitialized -Wunused -Wunused-local-typedefs -Wunused-parameter -Wunused-value -Wunused-variable -Wno-reorder -fno-rtti -fno-exceptions -std=c++11 -Winvalid-pch -Xclang -include-pch -Xclang D:/gitroot/vulkanjava/Vulkan-Samples/build/android_gradle/app/build-native/Debug/6s723z10/arm64-v8a/third_party/glslang/glslang/CMakeFiles/MachineIndependent.dir/cmake_pch.hxx.pch -Xclang -include -Xclang D:/gitroot/vulkanjava/Vulkan-Samples/build/android_gradle/app/build-native/Debug/6s723z10/arm64-v8a/third_party/glslang/glslang/CMakeFiles/MachineIndependent.dir/cmake_pch.hxx -fpch-preprocess -v -dD -E Android (9352603, based on r450784d1) clang version 14.0.7 (https://android.googlesource.com/toolchain/llvm-project 4c603efb0cca074e9238af8b4106c30add4418f6) Target: aarch64-none-linux-android30 Thread model: posix InstalledDir: C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/bin (in-process) "C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/bin/clang++.exe" -cc1 -triple aarch64-none-linux-android30 -E -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name compiler-file13038264060395318228 -mrelocation-model pic -pic-level 2 -fhalf-no-semantic-interposition -mframe-pointer=non-leaf -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu generic -target-feature +neon -target-feature +v8a -target-feature +fix-cortex-a53-835769 -target-abi aapcs -fallow-half-arguments-and-returns -debug-info-kind=standalone -dwarf-version=4 -debugger-tuning=gdb -v -ffunction-sections -fdata-sections -fcoverage-compilation-dir=D:/gitroot/vulkanjava/Vulkan-Samples/build/android_gradle/app/build-native/Debug/6s723z10/arm64-v8a -resource-dir C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/lib64/clang/14.0.7 -D ENABLE_HLSL -D ENABLE_OPT=0 -D GLSLANG_OSINCLUDE_UNIX -D ROOT_PATH_SIZE=37 -D VKB_DEBUG -I D:/gitroot/vulkanjava/Vulkan-Samples/build/android_gradle/app/build-native/Debug/6s723z10/arm64-v8a/include -D ANDROID -D _FORTIFY_SOURCE=2 -D DEBUG=0 -D ___CIDR_DEFINITIONS_END -isysroot C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot -internal-isystem C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1 -internal-isystem C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/lib64/clang/14.0.7/include -internal-isystem C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/local/include -internal-externc-isystem C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/aarch64-linux-android -internal-externc-isystem C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot/include -internal-externc-isystem C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include -Wformat -Wall -Wuninitialized -Wunused -Wunused-local-typedefs -Wunused-parameter -Wunused-value -Wunused-variable -Wno-reorder -Winvalid-pch -std=c++11 -fdeprecated-macro -fdebug-compilation-dir=D:/gitroot/vulkanjava/Vulkan-Samples/build/android_gradle/app/build-native/Debug/6s723z10/arm64-v8a -ferror-limit 19 -stack-protector 2 -fno-rtti -fno-signed-char -fgnuc-version=4.2.1 -dD -include-pch D:/gitroot/vulkanjava/Vulkan-Samples/build/android_gradle/app/build-native/Debug/6s723z10/arm64-v8a/third_party/glslang/glslang/CMakeFiles/MachineIndependent.dir/cmake_pch.hxx.pch -include D:/gitroot/vulkanjava/Vulkan-Samples/build/android_gradle/app/build-native/Debug/6s723z10/arm64-v8a/third_party/glslang/glslang/CMakeFiles/MachineIndependent.dir/cmake_pch.hxx -target-feature +outline-atomics -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o - -x c++ "C:\\Users\\voracity\\AppData\\Local\\Temp\\compiler-file13038264060395318228" clang -cc1 version 14.0.7 based upon LLVM 14.0.7git default target x86_64-w64-windows-gnu error: unable to read PCH file D:/gitroot/vulkanjava/Vulkan-Samples/build/android_gradle/app/build-native/Debug/6s723z10/arm64-v8a/third_party/glslang/glslang/CMakeFiles/MachineIndependent.dir/cmake_pch.hxx.pch: 'No such file or directory' ignoring nonexistent directory "C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot/include" #include "..." search starts here: #include ...> search starts here: D:/gitroot/vulkanjava/Vulkan-Samples/build/android_gradle/app/build-native/Debug/6s723z10/arm64-v8a/include C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1 C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/lib64/clang/14.0.7/include C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/local/include C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/aarch64-linux-android C:/Users/voracity/AppData/Local/Android/Sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include End of search list. 1 error generated. 

the same "unable to read" and "ignoring nonexistent directory" persists here. A thing to note tho which the person who originally made the post (perhaps it was the same for him he just didn't notice it) the file cmake_pch.hxx.pch doesn't exist. I have a cmake_pch.hxx.cxx and cmake_pch.hxx but not cmake_pch.hxx.pch and the include folder that can't be found is presumably found at sysroot/usr/include so I'm assuming that's fine but the .pch file doesn't exist.  Also someone said to try and remove cmake or remove it from the path and I removed it from the path. Unfortunately it didn't work.

**Comment by SaschaWillems** [comment]
Can you post the exact tool versions you are using?

- Android Studio
- Android SDK/NDK
- CMake
- Gradle

And the way you try to compile the samples (directly from Android Studio, command line, etc.)

This error might be caused by tooling that's outdated or too new.

**Comment by iexavl** [comment]
> Can you post the exact tool versions you are using?
> 
> * Android Studio
> * Android SDK/NDK
> * CMake
> * Gradle
> 
> And the way you try to compile the samples (directly from Android Studio, command line, etc.)
> 
> This error might be caused by tooling that's outdated or too new.

Android studio Giraffe 2022.3.1 Patch 4
ndk\25.2.9519653
cmake version 3.27.0-rc4
Gradle 7.5

For the way I build it at first when I opened the project it just tried to compile and it gave me this error. If I go to android studio Build->Refresh Linked C++ Projects it gives me the same error.

**Comment by GoldenSoju** [comment]
I have a similar problem in a similar setup.
```
Android Studio Hedgehog | 2023.1.1 Patch 2
ndk\25.2.9519653
cmake version 3.22.1
Gradle 7.5
```

First I wasn't able to get the `cmake_pch.hxx.pch` file, but it was created after I downloaded the Android SDK 30 sources (via sdk manager).
But still the app won't start (I tried with different emulators) and throws the error "Native code library failed to load".

**UPDATE:** It won't work on emulators but the app runs on my physical device.
